src/opm/io/eclipse/OutputStream.cpp

```diff
@@ -505,6 +528,220 @@ namespace Opm { namespace EclIO { namespace OutputStream {
 
 // =====================================================================
 
+namespace Opm { namespace EclIO { namespace OutputStream {
+
+namespace {
+    bool validUnitConvention(const SummarySpecification::UnitConvention uconv)
+    {
+        using UConv = SummarySpecification::UnitConvention;
+
+        return (uconv == UConv::Metric)
+            || (uconv == UConv::Field)
+            || (uconv == UConv::Lab)
+            || (uconv == UConv::Pvt_M);
+    }
+
+    int unitConvention(const SummarySpecification::UnitConvention uconv)
+    {
+        const auto unit = static_cast<int>(uconv);
+
+        if (! validUnitConvention(uconv)) {
+            throw std::invalid_argument {
+                "Invalid Unit Convention: " +
+                std::to_string(unit)
+            };
+        }
+
+        return unit;
+    }
+
+    int makeRestartStep(const SummarySpecification::RestartSpecification& restart)
+    {
+        return (restart.step >= 0) ? restart.step : -1;
+    }
+
+    std::vector<PaddedOutputString<8>>
+    restartRoot(const SummarySpecification::RestartSpecification& restart)
+    {
+        auto ret = std::vector<PaddedOutputString<8>>(9);
+
+        if (restart.root.empty()) {
+            return ret;
+        }
+
+        if (restart.root.size() > decltype(ret.size()){8 * 9}) {
```

Comment:
I would say this is warning - and no embedded restart information - rather than a full blown exception.

Reply:
Okay.